local_planner/src/nodes/common.cpp

```diff
@@ -131,3 +131,55 @@ void normalize(geometry_msgs::Point &p) {
     p.z = 0;
   }
 }
+
+double velocitySigmoid(double max_vel, double min_vel, double slope,
+                       double v_old, double elapsed) {
+  max_vel += 0.05;
+  min_vel -= 0.05;
+  v_old -= min_vel;
+  double t_old = -1.0 / slope * log((max_vel - min_vel) / v_old - 1.0);
+  double t_new = t_old + elapsed;
+  double speed = min_vel + (max_vel - min_vel) / (1.0 + exp(-slope * t_new));
+  return speed;
+}
+
+double velocityLinear(double max_vel, double min_vel, double slope,
+                      double v_old, double elapsed) {
+  v_old -= min_vel;
+  double t_old = v_old / slope;
+  double t_new = t_old + elapsed;
+  double speed = min_vel + t_new * slope;
+  return speed;
+}
+
+double getAngularVelocity(double new_yaw, double curr_yaw) {
+  while (new_yaw > M_PI) {
+    new_yaw -= M_PI;
```

Comment:
Is this correct? Why not `new_yaw -= 2*M_PI` ? I haven't completely thought it through yet, but conditionally rotating by 180 degrees is counter-intuitive at best. (In that case, clearly explain with comments why it is done)

Maybe refactor this logic into a dedicated function like `double wrapToPi(const double& input)`